pkg/provisioner/provisioner.go

```diff
@@ -99,4 +99,7 @@ type Result struct {
 	RequeueAfter time.Duration
 	// Any error message produced by the provisioner.
 	ErrorMessage string
+	// ErrorCount to keep track of the number of times an operation
+	// has encountered errors.
+	ErrorCount int
```

Comment:
So, based on this https://github.com/metal3-io/baremetal-operator/pull/412/commits/e87a2e8063b2ccd4e60a393d7684457389139c51#r522255159 comment, it is no longer necessary to add this ErrorCount here.